packages/block-editor/src/components/block-list/block-list-item.native.js

```diff
@@ -89,6 +93,50 @@ export class BlockListItem extends Component {
 		];
 	}
 
+	componentDidUpdate() {
+		this.scrollToSelectedBlockIfItsNotVisible();
+	}
+
+	scrollToSelectedBlockIfItsNotVisible() {
+		const { isSelected } = this.props;
+
+		if ( this.blockRef.current ) {
+			// first we are measuring the layout to determine where {x,y} within the List Component the current Block based View is located.
+			this.blockRef.current.measureLayout(
+				findNodeHandle( this.props.listRef.current ),
+				( x, y ) => {
+					// once we have the {x,y} of that View we can then determine if it is currently visible.
+					this.blockRef.current.measure(
+						( _x, _y, blockWidth, blockHeight, px, py ) => {
+							const window = Dimensions.get( 'window' );
+
+							if ( isSelected ) {
```

Comment:
It looks like this can be moved up to the beginning of `scrollToSelectedBlockIfItsNotVisible` to a) avoid execution reaching here if there's no selected block and b) reduce the number of levels of indentation.
```

  scrollToSelectedBlockIfItsNotVisible() {
    const { isSelected } = this.props;
    if (!isSelected) return;
    ...
    
```